codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -481,8 +473,23 @@ let updateChainFetcher = (chainFetcherUpdate, ~state, ~chain) => {
 
 let actionReducer = (state: t, action: action) => {
   switch action {
-  | SetFetchStateCurrentBlockHeight(chain, currentBlockHeight) =>
-    state->handleSetCurrentBlockHeight(~chain, ~currentBlockHeight)
+  | StartWaitingForNewBlock({chain}) => updateChainFetcher(currentChainFetcher => {
+      ...currentChainFetcher,
+      isWaitingForNewBlock: true,
+    }, ~chain, ~state)
+  | FinishWaitingForNewBlock({chain, currentBlockHeight}) => (
+      {
+        ...state,
+        chainManager: {
+          ...state.chainManager,
+          chainFetchers: state.chainManager.chainFetchers->ChainMap.update(chain, chainFetcher => {
+            ...chainFetcher->updateChainFetcherCurrentBlockHeight(~currentBlockHeight),
+            isWaitingForNewBlock: false,
+          }),
+        },
+      },
```

Comment:
Looks like this can use the `updateChainFetcher` helper like above. In fact looks like these two actions could be combined to pass "isWaitingForNewBlock" and just set that value.

Reply:
Nvmnd, see it includes current blockheight 👍🏼 but can still use the helper fn